### DISCUSSION TITLE
Bug with Dynamic Theming after recent update

### DISCUSSION THREAD

**Discussion by J43git** [sadellie/unitto#10] [General]
I don't think people recognize this or know it exists, but this is a fantastic app. It is the best unit converter and now calculator that I've ever used. 

With this new update I was excited to try out the new features. I opened up the app, and saw purple buttons and purple accent which matched my wallpaper. But after I changed it to half pink half blue, the accent is blue but the buttons are still purple. Changed to pink wallpaper, pink accent, purple buttons.

**Comment by sadellie** [comment]
Hello, thank you for the kind words.

Dynamic theming works strange in some cases. I am aware of it and currently trying to find a solution (maybe more settings for theming, i.e. "choose your colors").

Could you please tell me which Android Version are you on?

Also, it would be great if you shared the wallpaper that caused an issue (don't share if there is anything personal). 

Thank you for using Unitto.

**Reply by J43git** [reply]
![zv0Oism](https://user-images.githubusercontent.com/112982953/207719272-341a4d6e-221d-43d7-a517-44d48e12f622.jpg)

I'm using Android 10

**Reply by J43git** [reply]
Nice update but problem still persists. However, I don't want you to feel like I'm rushing you.

**Reply by sadellie** [reply]
Fixed in e5b7a7b6c2373401425c0f95e1b4c2ae25a393de. Will be in the next update.

NOTE: Dynamic Color is officially supported for devices with Android 12+. Older devices use an algorithm inspired by [Docile-Alligator/Infinity-For-Reddit](https://github.com/Docile-Alligator/Infinity-For-Reddit). It tries its best, but still can produce weird color scheme.

Thank you for reporting this bug!

**Reply by J43git** [reply]
![InCollage_20221229_212814912](https://user-images.githubusercontent.com/112982953/210035600-b0c78dc9-cbe6-484f-a56e-5e3c435b33c8.jpg)
It's nice to see it's been fixed! I think you made it even better as shown by this image above this text. I wish I can find out how to get the word out about this app because of how versatile it is though.
